core/src/main/java/io/questdb/cairo/mv/MatViewRefreshJob.java

```diff
@@ -503,18 +593,21 @@ private boolean insertAsSelect(
                     long commitTarget = batchSize;
                     long rowCount = 0;
 
-
                     intervalIterator.toTop(intervalStep);
                     long replacementTimestampLo = Long.MIN_VALUE;
                     long replacementTimestampHi = Long.MIN_VALUE;
 
                     while (intervalIterator.next()) {
-                        refreshExecutionContext.setRange(intervalIterator.getTimestampLo(), intervalIterator.getTimestampHi());
+                        refreshSqlExecutionContext.setRange(intervalIterator.getTimestampLo(), intervalIterator.getTimestampHi());
```

Comment:
Lo timestamp may be prior to the `minTs` from `findRefreshIntervals()` (see Alex' test data).